### PR TITLE
[BEAM-79] merge gearpump-runner into master

### PR DESCRIPTION
**Pull request by manuzhang**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by manuzhang** [comment]
R: @kennknowles

File: runners/gearpump/pom.xml [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+    <gearpump.version>0.8.4</gearpump.version>
+  </properties>
+
+  <profiles>
+    <profile>
+      <id>local-validates-runner-tests</id>
+      <!-- TODO: once on master branch, move to postcommit only -->
+      <activation><activeByDefault>true</activeByDefault></activation>
```
Comment (kennknowles) [inline runners/gearpump/pom.xml]: OK, now this should not be active by default. It should run only on postcommit.

File: examples/java/pom.xml [hunk 2]
```diff
@@ -89,6 +89,18 @@
       </dependencies>
     </profile>
 
+    <!-- Include the Apache Gearpump (incubating) runner with -P gearpump-runner -->
+    <profile>
+      <id>gearpump-runner</id>
```
Comment (kennknowles) [inline examples/java/pom.xml]: Take a look in this file for the `jenkins-precommit` bits. You can add an execution for the Gearpump runner.

**Comment by manuzhang** [comment]
@kennknowles split this into #3636 and #3637. I will leave it open for a while since it is referenced on mailing list

**Comment by kennknowles** [comment]
Great. Those PRs look just right. Fine to leave this open - but links will keep working even when close.

**Pull request closed without merging**